src/Routing/RouteBuilder.php

```diff
@@ -932,13 +926,12 @@ public function plugin(string $name, $options = [], ?callable $callback = null)
      */
     public function scope(string $path, $params, $callback = null)
     {
-        if (is_callable($params)) {
+        if (!is_array($params)) {
             $callback = $params;
             $params = [];
         }
         if (!is_callable($callback)) {
-            $msg = 'Need a callable function/object to connect routes.';
-            throw new InvalidArgumentException($msg);
+            throw new InvalidArgumentException('Need a callable function/object to connect routes.');
```

Comment:
Could we include what type we got here to help the developer find where they went wrong?